servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -327,11 +361,13 @@ public boolean isDirectory(long id) {
    *
    * @param path the path to get the file id for
    * @return the file id for a given path, or -1 if there is no file at that path
+   * @throws AccessControlException if permission checking fails
    */
-  public long getFileId(TachyonURI path) {
+  public long getFileId(TachyonURI path) throws AccessControlException {
     synchronized (mInodeTree) {
       Inode inode;
       try {
+        checkPermission(FileSystemAction.READ, path, false);
```

Comment:
It would be good to specify which permissions are needed for each method, and which methods check permissions (for example, `getFileId` may be used internally which would implicitly make the method require `READ` rights).